scripts/split_repo_migration/git-migrate-history.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+if [ $# -lt 4 ]
+then
+  echo "Usage: $0 <source-repo> <target-repo> <source-path> <target-path> [folders,to,skip] [files,to,keep] [branch-name]"
+  exit 1
```

Comment:
If the user sources this script rather than executing it, this will kill their shell. I suggest at the top of the file:
- either enforcing that the script is being executed, or
- defining `$EXIT` and using it in place of `exit`, like so:
  ```
  [[ "${BASH_SOURCE[0]}" != "${0}" ]] && EXIT=return || EXIT=exit
  ```

Reply:
Nice catch! thanks.